### PR TITLE
Update Expecto version to 7.0.0

### PR DESCRIPTION
**Pull request by TheAngryByrd**
Based off work from https://github.com/atlemann/dotnet-test-workaround we can use a set of frameworkoverrides provided by the Don https://github.com/dotnet/sdk/issues/335#issuecomment-368669050 that allow building netfull on mono with dotnet core tooling.  

Had to bump expecto version to 7.0 otherwise was getting an error 'executor://yolodev/expecto': Method 'Expecto.Impl.get_testFromAssembly' not found.

### CODE REVIEW

**Comment by TheAngryByrd** [comment]
>>> /home/travis/build/YoloDev/YoloDev.Expecto.TestSdk/test/Sample.Test/Sample.Test.fsproj : error NU1605: Detected package downgrade: FSharp.Core from 4.3.2 to 4.2.0. Reference the package directly from the project to select a different version.  

So we have two options.  Bumping the dotnet sdk to a higher version or actually referencing fsharp.core in the project files.  What do you think is more appropriate?

**Comment by Alxandr** [comment]
What is the purpose of this PR? Updating Expecto? Or are you saying that `YoloDev.Expecto.TestSdk` as is does not run on mono? Cause as far as I can see, your PR will not have any effect on that.

File: netfx.props [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    <FrameworkPathOverride Condition="'$(BaseFrameworkPathOverrideForMono)' != '' AND '$(TargetFramework)' == 'net46'">$(BaseFrameworkPathOverrideForMono)/4.6-api</FrameworkPathOverride>
+    <FrameworkPathOverride Condition="'$(BaseFrameworkPathOverrideForMono)' != '' AND '$(TargetFramework)' == 'net461'">$(BaseFrameworkPathOverrideForMono)/4.6.1-api</FrameworkPathOverride>
+    <FrameworkPathOverride Condition="'$(BaseFrameworkPathOverrideForMono)' != '' AND '$(TargetFramework)' == 'net462'">$(BaseFrameworkPathOverrideForMono)/4.6.2-api</FrameworkPathOverride>
+    <FrameworkPathOverride Condition="'$(BaseFrameworkPathOverrideForMono)' != '' AND '$(TargetFramework)' == 'net47'">$(BaseFrameworkPathOverrideForMono)/4.7-api</FrameworkPathOverride>
+    <FrameworkPathOverride Condition="'$(BaseFrameworkPathOverrideForMono)' != '' AND '$(TargetFramework)' == 'net471'">$(BaseFrameworkPathOverrideForMono)/4.7.1-api</FrameworkPathOverride>
+    <EnableFrameworkPathOverride Condition="'$(BaseFrameworkPathOverrideForMono)' != ''">true</EnableFrameworkPathOverride>
+
+  </PropertyGroup>
+</Project>
```
Comment (Alxandr) [inline netfx.props]: Lacking endline at end of file.

File: test/Directory.Build.props [hunk 2]
```diff
@@ -10,4 +10,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(TestSdkVersion)" />
     <ProjectReference Include="$(MSBuildThisFileDirectory)\..\src\YoloDev.Expecto.TestSdk\YoloDev.Expecto.TestSdk.fsproj" />
   </ItemGroup>
+  <ItemGroup>
+    <DotNetCliToolReference Include="Microsoft.DotNet.Watcher.Tools" Version="2.0.0" />
```
Comment (Alxandr) [inline test/Directory.Build.props]: While I agree that having the watcher might be nice, if it's to be added that should be a different PR.

Comment (TheAngryByrd) [reply test/Directory.Build.props]: Will remove

File: netfx.props (hunk 1 above)
Comment (TheAngryByrd) [reply netfx.props]: Will fix

**Comment by TheAngryByrd** [comment]
As it currently stands YoloDev.Expecto.TestSdk doesn't work on mono.  This adds `netfx.props` which allows the projects to build on mono and subsequently `dotnet test` will work on macOS/linux. As stated I had to bump expecto version because that error was displaying for tests on mono specifically.

**Comment by Alxandr** [comment]
So what you're saying is that bumping Expecto makes it work on mono, and the `netfx.props` is used for enabling building on mono (just so we can test on mono on travis and co)?

**Comment by TheAngryByrd** [comment]
Yes

**Comment by Alxandr** [comment]
Gotcha. I think adding a minimum requirement to FSharp.Core is probably ok. Also, can you rename `netfx.props` to something more descriptive? And that starts with a capital letter. And probably also move it into the `build` folder. Something like `Mono.DotNetCore.props` or similar maybe?

**Comment by MNie** [comment]
@Alxandr any update about that pr? Since I saw there is a problem in Expecto.Template when Expecto > 7.0 is using along with Yolo.TestDev.

**Comment by TheAngryByrd** [comment]
@MNie I'll be fixing it today.

**Comment by TheAngryByrd** [comment]
And now of course I'm getting an error I can't reproduce locally on osx or ubuntu 😿 

```
e/travis/build/YoloDev/YoloDev.Expecto.TestSdk/test/Sample.Test/bin/Debug/net462/Sample.Test.exe(.NETFramework,Version=v4.6.2)
Microsoft (R) Test Execution Command Line Tool Version 15.3.0-preview-20170628-02
Copyright (c) Microsoft Corporation.  All rights reserved.
Starting test execution, please wait...
Unhandled Exception: System.AggregateException: One or more errors occurred. (Permission denied) ---> System.ComponentModel.Win32Exception: Permission denied
   at Interop.Sys.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Int32& lpChildPid, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.ProcessHelper.LaunchProcess(String processPath, String arguments, String workingDirectory, IDictionary`2 envVariables, Action`2 errorCallback, Action`1 exitCallBack)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager.LaunchHost(TestProcessStartInfo testHostStartInfo)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager.<>c__DisplayClass34_0.<LaunchTestHostAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager.<LaunchTestHostAsync>d__34.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.InitializeExtensions(IEnumerable`1 sources)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyExecutionManager.Initialize()
   at Microsoft.VisualStudio.TestPlatform.Client.TestPlatform.CreateTestRunRequest(TestRunCriteria testRunCriteria, ProtocolConfig protocolConfig)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.RunTests(TestRunCriteria testRunCriteria, ITestRunEventsRegistrar testRunEventsRegistrar, ProtocolConfig protocolConfig)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers.TestRequestManager.RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLauncher testHostLauncher, ITestRunEventsRegistrar testRunEventsRegistrar, ProtocolConfig protocolConfig)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.RunTestsArgumentExecutor.RunTests(IEnumerable`1 sources)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.RunTestsArgumentExecutor.Execute()
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.ExecuteArgumentProcessor(IArgumentProcessor processor, Int32& exitCode)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Executor.Execute(String[] args)
   at Microsoft.VisualStudio.TestPlatform.CommandLine.Program.Main(String[] args)
Build FAILED.
    0 Warning(s)
    0 Error(s)
Time Elapsed 00:00:36.01
```

**Comment by Alxandr** [comment]
Hmm. It's passing on AppVeyor though. I'm going to accept it, and do a release. And if it turns out to be a problem Ima have to add some more tests and try it out on a windows box.

**Comment by Alxandr** [comment]
v0.5.0 is live on NuGet.
